modules/user_accounts/ajax/rejectUser.php

```diff
@@ -1,122 +1,76 @@
+ */
 function _hasPerm()
 {
     $user =& User::singleton();
     if (isset($user)) {
-        if (!$user->hasPermission('user_accounts')) {
-            return false;
-        }
-        return true;
+        return $user->hasPermission('user_accounts');
     }
     return false;
 }
 
-    /**
-    * Query database on UserID to check that password has doesn't
-    * exist and account is still pending approval, to ensure the
-    * users with account activity can't be rejected by setting
-    * their status to pending.
-    *
-    * @param string $identifier of account to verify
-    *
-    * @return boolean true if Password_hash is null and
-    * Pending_approval is Yes, false otherwsie
-    *
-    * @throws Loris exception if the userID isn't found in the database.
-    */
-function _canRejectAccount($identifier)
+/**
+ * Send DB query to remove user based on userID, checking that the
+ * user sending the request has admin permissions and the account
+ * can be removed (no password hash => has never had an activity on
+ * Loris, and is pending).
+ *
+ * @param string $userID of account to reject
+ *
+ * @return void
+ */
+function _rejectUser($userID)
 {
-    $DB = Database::singleton();
-    try{
-        $fields =  $DB->pselect(
-            "SELECT Password_hash, Pending_approval
-                       FROM users
-                       WHERE UserID=:id",
-            array("id" => $identifier)
-        );
-        if (is_null($fields[0]['Password_hash'])
-            && $fields[0]['Pending_approval']=="Y"
-        ) {
-            return true;
-        }
-        return false;
-    } catch (DatabaseException $e) {
-        throw new LorisException("User not found in database", USER_NOT_FOUND);
-        exit(1);
-    }
-}
-
-
-    /**
-    * Send DB query to remove user based on userID, checking that the
-    * user sending the request has admin permissions and the account
-    * can be removed (no password hash => has never had an activity on
-    * Loris, and is pending).
-    *
-    * @return void
-    */
-    // @codingStandardsIgnoreStart
-    function reject_user($identifier)
-    {
-    // @codingStandardsIgnoreEnd
-        $DB      = Database::singleton();
-        $config  = NDB_Config::singleton();
-        $baseURL = $config->getSetting('url');
-
-        $redirect = $baseURL . "/user_accounts/";
-
+    $DB       = \Database::singleton();
+    $config   = \NDB_Config::singleton();
+    $baseURL  = $config->getSetting('url');
+    $redirect = $baseURL . "/user_accounts/";
     if (!_hasPerm()) {
         throw new LorisException(
             "You do not have the correct permissions for this 
             operation (need admin or user accounts)",
             INCORRECT_PERMISSION
         );
-        exit(1);
+    } else if (!UA\Edit_User::canRejectAccount($userID)) {
+        throw new LorisException(
+            "This account is active and cannot be rejected",
```

Comment:
403 Forbidden or 409 Conflict not sure. @driusan ?

Reply:
I think forbidden, conflict is for things like multiple people editing at the same time.